.github/workflows/main.yml

```diff
@@ -3,19 +3,35 @@ name: Docker image CI
 on:
   push:
     branches: ['master']
-  workflow_dispatch:
+  
+env:
+  IMAGE_TAG: latest
+  REPO_URL: ${{ github.repository }}
+    
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+  
+      - name: Checkout code 
+        uses: actions/checkout@v3
+    
+      - name: Extract branch name
+        id: extract_branch
+        run: echo "BRANCH_NAME=$(basename ${{ github.ref }})" >> $GITHUB_ENV
```

Comment:
This is the step the pull the branch it is built from.

Reply:
There was an env variable called github.base_ref but some reason it did not work. So I have to adopt this workaround